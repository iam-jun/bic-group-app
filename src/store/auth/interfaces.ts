import {CognitoUser, ISignUpResult} from 'amazon-cognito-identity-js';
// Actiontype
export interface ISignIn {
  email: string;
  password: string;
}

export interface ISignUp {
  username: string;
  email: string;
  password: string;
}

export interface IForgotPassword {
  email: string;
  callback: () => void;
}

export interface IForgotPasswordRequest {
  code: string;
  email: string;
  password: string;
}

export interface IUser extends CognitoUser {}

export interface ISignUpResponse extends ISignUpResult {}
