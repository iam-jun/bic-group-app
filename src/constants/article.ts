export enum EditArticleErrorType {
  SERIES_DENIED = 'SERIES_DENIED',
  GROUPS_DENIED = 'GROUPS_DENIED',
  TAGS_DENIED = 'TAGS_DENIED',
}

export const EMPTY_ARTICLE_CONTENT = [{ type: 'p', children: [{ text: '' }] }];
