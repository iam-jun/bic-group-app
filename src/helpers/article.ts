import i18next from 'i18next';
import useModalStore from '~/store/modal';
import ImageUploader, { IGetFile, IUploadParam } from '~/services/imageUploader';

export const uploadImage = async ({ file, onSuccess }: {
  file: any, onSuccess: (file: IGetFile) => void
}) => {
  const onError = (error) => {
    const content = typeof error === 'string' ? error : i18next.t('post:error_upload_photo_failed');
    useModalStore.getState().actions.showToast({ content });
  };

  const uploadType = 'post_image';
  const param: IUploadParam = {
    uploadType,
    file,
    onSuccess,
    onError,

  };
  try {
    await ImageUploader.getInstance().upload(param);
  } catch (error) {
    onError(error);
  }
};
