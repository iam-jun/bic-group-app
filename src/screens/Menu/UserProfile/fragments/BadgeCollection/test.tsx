import React from 'react';
import {
  View, Dimensions, FlatList, StyleSheet,
} from 'react-native';
import { ExtendedTheme, useTheme } from '@react-navigation/native';
import Avatar from '~/baseComponents/Avatar';
import Icon from '~/baseComponents/Icon';
import spacing, { borderRadius } from '~/theme/spacing';

const Grid = ({ data }: {data: any[]}) => {
  const deviceWidth = Dimensions.get('window').width;

  const itemWidth = 48; // Constant item width
  const numColumns = Math.floor((deviceWidth - 10 * 2) / (itemWidth + 6 * 2));
  const rowWidth = numColumns * (itemWidth + 6 * 2);
  const paddingLeft = (deviceWidth - (10 * 2) - rowWidth) / 2;

  const theme: ExtendedTheme = useTheme();
  const { colors } = theme;
  const styles = themeStyles(theme);

  const isSelected = false;

  const renderGridItem = ({ item, index }: any) =>
    // Render each grid item here
    (
      <View
        key={index}
        style={{
          margin: 6,
          borderWidth: isSelected ? 2 : 1,
          borderColor: isSelected ? colors.purple50 : colors.neutral5,
          borderRadius: borderRadius.pill,
        }}
      >
        <Avatar.Medium
          isRounded
          source={{ uri: item?.iconUrl }}
        />
        {isSelected
          ? (
            <View style={styles.iconChose}>
              <Icon size={10} icon="Check" tintColor={colors.white} />
            </View>
          )
          : null}
      </View>
    );

  return (
    <View style={{
      flex: 1,
      justifyContent: 'center',
      alignItems: 'center',
      paddingHorizontal: 10,
      paddingBottom: 10,
    }}
    >

      <View style={{
        // paddingLeft,
      }}
      >
        <FlatList
          data={data}
          renderItem={renderGridItem}
          keyExtractor={(item, index) => index.toString()}
          numColumns={numColumns}
        />
      </View>
    </View>
  );
};

const themeStyles = (theme: ExtendedTheme) => {
  const { colors } = theme;

  return StyleSheet.create({
    container: {
      flex: 1,
      paddingTop: spacing.padding.large,
    },
    iconChose: {
      backgroundColor: colors.purple50,
      borderRadius: borderRadius.pill,
      height: 20,
      width: 20,
      alignItems: 'center',
      justifyContent: 'center',
      position: 'absolute',
      zIndex: 1,
      right: 0,
    },
  });
};

export default Grid;
