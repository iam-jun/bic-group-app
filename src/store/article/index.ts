import {
  resetStore,
  createStore,
} from '~/store/utils';
import { IArticlesState } from './Interface';
import getArticleDetail from './actions/getArticleDetail';

const initialState = {
  items: {},
  requestings: {},

};

const articleStore = (set, get) => ({
  ...initialState,
  actions: {
    getArticleDetail: getArticleDetail(set, get),
  },
  reset: () => resetStore(initialState, set),
});

const useArticleStore = createStore<IArticlesState>(articleStore);

export default useArticleStore;
