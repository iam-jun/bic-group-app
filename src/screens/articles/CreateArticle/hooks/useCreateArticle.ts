import { isEmpty, isEqual } from 'lodash';
import {
  useEffect, useMemo, useRef, useState,
} from 'react';
import { Keyboard } from 'react-native';

import moment from 'moment';
import { useShallow } from '~/store/utils';
import useMentionInputStore from '~/beinComponents/inputs/MentionInput/store';
import IMentionInputState from '~/beinComponents/inputs/MentionInput/store/Interface';
import {
  IEditArticleAudience,
  IEditArticleData,
  IParamsValidateSeriesTags,
  IPayloadPublishDraftArticle,
  IPayloadPutEditArticle,
} from '~/interfaces/IArticle';
import { withNavigation } from '~/router/helper';
import { getAudienceIdsFromAudienceObject, isEmptyContent } from '~/screens/articles/CreateArticle/helper';
import useCreateArticleStore from '~/screens/articles/CreateArticle/store';
import { getMentionsFromContent } from '~/helpers/post';
import usePostsStore from '~/store/entities/posts';
import postsSelector from '~/store/entities/posts/selectors';
import { useBaseHook } from '~/hooks';

import { rootNavigationRef } from '~/router/refs';
import articleStack from '~/router/navigator/MainStack/stacks/articleStack/stack';
import useDraftArticleStore from '~/screens/Draft/DraftArticle/store';
import useScheduleArticlesStore from '~/screens/YourContent/components/ScheduledArticles/store';
import useModalStore from '~/store/modal';
import { PostStatus } from '~/interfaces/IPost';

interface IHandleSaveOptions {
  isShowLoading?: boolean;
  isNavigateBack?: boolean;
  isShowToast?: boolean;
  shouldValidateSeriesTags?: boolean;
  onSuccess?: () => void;
  titleAlert?: string;
}

const navigation = withNavigation(rootNavigationRef);

export interface IUseEditArticle {
  articleId: string;
  handleSaveAudienceError?: (listIdAudiences: string[]) => void;
}

const useCreateArticle = ({
  articleId,
}: IUseEditArticle) => {
  const article = usePostsStore(postsSelector.getPost(articleId, {}));

  const actions = useCreateArticleStore((state) => state.actions);

  const data = useCreateArticleStore((state) => state.data, useShallow) || {};
  const loading = useCreateArticleStore((state) => state.loading);
  const isValidating = useCreateArticleStore((state) => state.isValidating);
  const isDraft = useCreateArticleStore((state) => state.isDraft);
  const publishedAt = useCreateArticleStore((state) => state.schedule.publishedAt);

  const { showAlert } = useModalStore((state) => state.actions);

  const [isShowToastAutoSave, setShowToastAutoSave] = useState<boolean>(false);

  const refStopTyping = useRef(null);
  const refTypingConstantly = useRef(null);

  const tempMentions = useMentionInputStore(
    (state: IMentionInputState) => state.tempSelected,
  );

  const groupIds = useMemo(
    () => data.audience?.groupIds?.join?.(','),
    [data.audience],
  );

  const { t } = useBaseHook();

  const isValidScheduleTime = () => moment(publishedAt).isSameOrAfter(moment());

  // auto save for draft article, so no need to check if content is empty
  const isDraftContentUpdated
    = article.content !== data.content;

  const isHasChange = () => {
    // self check at src/screens/articles/CreateArticle/screens/CreateArticleContent/index.tsx
    // const isContentUpdated = article.content !== data.content && !isEmptyContent();
    const isSummaryUpdated = article.summary !== data.summary;
    const isTitleUpdated = article.title !== data.title;
    const isCategoriesUpdated
      = !isEqual(article.categories, data.categories);
    // self check at src/screens/articles/CreateArticle/screens/CreateArticleAudience/index.tsx
    // const isAudienceUpdated = !isEqual(getAudienceIdsFromAudienceObject(article.audience), data.audience)
    // && !(isEmpty(data.audience?.groupIds) && isEmpty(data.audience?.userIds));
    const isCoverMediaUpdated
      = article.coverMedia?.id !== data.coverMedia?.id;
    const isSeriesUpdated = !isEqual(article?.series, data.series);
    const isTagsUpdated = !isEqual(article?.tags, data.tags);

    return (
      isTitleUpdated
      || isSummaryUpdated
      || isCategoriesUpdated
      || isCoverMediaUpdated
      || isSeriesUpdated
      || isTagsUpdated
    );
  };

  const isValidForSave = () => {
    // self check at src/screens/articles/CreateArticle/screens/CreateArticleContent/index.tsx
    // const isContentUpdated = article.content !== data.content && !isEmptyContent();
    const isSummaryUpdated = article.summary !== data.summary;
    const isTitleUpdated = article.title !== data.title && !isEmpty(data.title?.trim());
    const isCategoriesUpdated
      = !isEqual(article.categories, data.categories)
      && !isEmpty(data.categories);
    // self check at src/screens/articles/CreateArticle/screens/CreateArticleAudience/index.tsx
    // const isAudienceUpdated = !isEqual(getAudienceIdsFromAudienceObject(article.audience), data.audience)
    // && !(isEmpty(data.audience?.groupIds) && isEmpty(data.audience?.userIds));
    const isCoverMediaUpdated
      = article.coverMedia?.id !== data.coverMedia?.id
      && !isEmpty(data.coverMedia);
    const isSeriesUpdated = !isEqual(article?.series, data.series);
    const isTagsUpdated = !isEqual(article?.tags, data.tags);

    return (
      isTitleUpdated
      || isSummaryUpdated
      || isCategoriesUpdated
      || isCoverMediaUpdated
      || isSeriesUpdated
      || isTagsUpdated
    );
  };

  const getValidButtonPublish = () => {
    const isTitleValid = !isEmpty(data.title);
    const isContentValid = !isEmptyContent(data.content);
    const isCategoriesValid = !isEmpty(data.categories);
    const isCoverValid = !isEmpty(data.coverMedia);
    const isAudienceValid = !(
      isEmpty(data.audience?.groupIds) && isEmpty(data.audience?.userIds)
    );

    return (
      isTitleValid
      && isContentValid
      && isCategoriesValid
      && isCoverValid
      && isAudienceValid
    );
  };

  const initEditStoreData = () => {
    const {
      id,
      title,
      content,
      audience: audienceObject,
      mentions,
      summary,
      categories,
      coverMedia,
      series,
      tags,
      status,
      publishedAt,
    } = article;
    const audienceIds: IEditArticleAudience
      = getAudienceIdsFromAudienceObject(audienceObject);
    const data: IEditArticleData = {
      id,
      title,
      content,
      audience: audienceIds,
      mentions,
      summary,
      categories,
      coverMedia,
      series,
      tags,
    };
    actions.setData(data);
    const isDraft = [PostStatus.DRAFT, PostStatus.WAITING_SCHEDULE, PostStatus.SCHEDULE_FAILED].includes(status);
    actions.setIsDraft(isDraft);
    if (isDraft) {
      actions.setPublishedAt(publishedAt || '');
    }
  };

  useEffect(() => {
    if (!isEmpty(article)) {
      initEditStoreData();
    }
  }, [article]);

  const enableButtonSave = isValidForSave();
  const validButtonPublish = getValidButtonPublish();
  const isChanged = isHasChange();

  const handleContentChange = (newContent: string) => {
    actions.setContent(newContent);
  };

  const handleTitleChange = (newTitle: string) => {
    actions.setTitle(newTitle);
  };

  const resetPublishedAt = () => {
    actions.setPublishedAt(article?.publishedAt || '');
  };

  const updateMentions = () => {
    const newMentions = getMentionsFromContent(data.content, tempMentions);
    actions.setMentions(newMentions);
  };

  const handleAudiencesChange = (newAudiences: IEditArticleAudience) => {
    actions.setAudience(newAudiences);
  };

  const showToastAutoSave = () => {
    setShowToastAutoSave(true);
    setTimeout(() => {
      setShowToastAutoSave(false);
    }, 2000);
  };

  const debounceStopTyping = () => {
    clearTimeout(refStopTyping.current);
    refStopTyping.current = setTimeout(() => {
      clearTimeout(refTypingConstantly.current);
      refTypingConstantly.current = null;
      handleSave({ isNavigateBack: false, isShowLoading: false, isShowToast: false });
      showToastAutoSave();
    }, 500);
  };

  const debounceTypingConstantly = () => {
    if (!refTypingConstantly.current) {
      refTypingConstantly.current = setTimeout(() => {
        handleSave({ isNavigateBack: false, isShowLoading: false, isShowToast: false });
        refTypingConstantly.current = null;
        showToastAutoSave();
      }, 5000);
    }
  };

  useEffect(() => {
    // only auto save for draft article
    if (isDraft && isDraftContentUpdated) {
      debounceStopTyping();
      debounceTypingConstantly();
    }
  }, [data.content]);

  useEffect(() => () => {
    actions.setIsDraft(false);
  }, []);

  const validateSeriesTags = (onSuccess: (response) => void, onError: (error) => void) => {
    const dataUpdate = useCreateArticleStore.getState().data;
    const validateParams: IParamsValidateSeriesTags = {
      groups: dataUpdate?.audience?.groupIds || [],
      series: dataUpdate?.series?.map?.((item) => item.id) || [],
      tags: dataUpdate?.tags?.map?.((item) => item.id) || [],
    };
    actions.validateSeriesTags(validateParams, onSuccess, onError);
  };

  const handleSave = (options?: IHandleSaveOptions) => {
    Keyboard.dismiss();
    const {
      isNavigateBack = true, isShowLoading = true, isShowToast = true, shouldValidateSeriesTags, onSuccess, titleAlert,
    } = options || {};
    updateMentions();

    // get data directly from store instead of hook
    // because in case we set store and immediately call handleSave
    // useCreateArticle hook is still not updated at that time
    // so handleSave will hold the old data instead of the new data
    const dataUpdate = useCreateArticleStore.getState().data;
    const putEditArticleParams = {
      articleId,
      data: dataUpdate,
      isNavigateBack,
      isShowToast,
      isShowLoading,
      onSuccess,
    } as IPayloadPutEditArticle;

    if (shouldValidateSeriesTags) {
      const onSuccess = () => actions.putEditArticle(putEditArticleParams);
      const onError = (error) => {
        actions.handleSaveError(error, () => handleSave(options), titleAlert);
      };
      validateSeriesTags(onSuccess, onError);
    } else {
      actions.putEditArticle(putEditArticleParams);
    }
  };

  const handlePublish = () => {
    if (!validButtonPublish) return;

    const goToArticleDetail = () => {
      navigation.replace(articleStack.articleDetail, { articleId: data.id });
    };

    const onHandleSaveErrorDone = () => handleSave({
      isNavigateBack: false,
      isShowLoading: true,
      isShowToast: false,
      onSuccess: () => handlePublish(),
    });

    const payload: IPayloadPublishDraftArticle = {
      draftArticleId: data.id,
      onSuccess: () => {
        goToArticleDetail();
        useScheduleArticlesStore.getState().actions.getScheduleArticles({ isRefresh: true });
      },
      onError: (error) => actions.handleSaveError(error, onHandleSaveErrorDone),
    };
    useDraftArticleStore.getState().actions.publishDraftArticle(payload);
  };

  const handleSchedule = () => {
    if (!validButtonPublish) return;

    if (!isValidScheduleTime()) {
      actions.setErrorScheduleSubmiting(t('article:fail_schedule'));
      return;
    }

    actions.scheduleArticle();
  };

  const handleBack = (shouldShowAlert = false) => {
    if (isChanged || shouldShowAlert) {
      Keyboard.dismiss();
      showAlert({
        title: t('discard_alert:title'),
        content: t('discard_alert:content'),
        cancelBtn: true,
        cancelLabel: t('common:btn_discard'),
        confirmLabel: t('common:btn_stay_here'),
        onCancel: () => {
          initEditStoreData();
          navigation.goBack();
        },
      });
      return;
    }
    navigation.goBack();
  };

  return {
    loading,
    isValidating,
    isShowToastAutoSave,
    enableButtonSave,
    validButtonPublish,
    title: data.title,
    content: data.content,
    groupIds,
    handleTitleChange,
    handleContentChange,
    handleSave,
    handleBack,
    handleAudiencesChange,
    handlePublish,
    handleSchedule,
    validateSeriesTags,
    resetPublishedAt,
  };
};

export default useCreateArticle;
