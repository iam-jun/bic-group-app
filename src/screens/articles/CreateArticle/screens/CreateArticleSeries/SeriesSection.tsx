import React, { FC } from 'react';
import { View, StyleSheet } from 'react-native';
import { isEmpty } from 'lodash';
import { useBaseHook } from '~/hooks';
import useCreateArticleStore from '../../store';
import PreviewSection from '../../components/PreviewSection';
import { spacing } from '~/theme';
import Tag from '~/baseComponents/Tag';

type SeriesSectionProps = {
  onPress: () => void;
};

const SeriesSection: FC<SeriesSectionProps> = ({ onPress }) => {
  const { t } = useBaseHook();
  const styles = createStyle();

  const audience = useCreateArticleStore((state) => state.data.audience);
  const series = useCreateArticleStore((state) => state.data.series);

  const hasAudience = !(isEmpty(audience?.groupIds) && isEmpty(audience?.userIds));

  const renderContent = () => {
    if (!series || series.length === 0) return null;

    return (
      <View style={styles.tagContainer}>
        {series.map((item) => (
          <View style={styles.tagView} key={`tagview-${item.id}`}>
            <Tag
              style={styles.tag}
              type="neutral"
              size="medium"
              label={item.title}
              textProps={{ numberOfLines: 1 }}
            />
          </View>
        ))}
      </View>
    );
  };

  return (
    <View>
      <PreviewSection
        title={t('article:text_option_edit_series')}
        optional
        onPress={onPress}
        placeholder={t('article:text_add_series')}
        content={renderContent()}
        disabled={!hasAudience}
      />
    </View>
  );
};

const createStyle = () => StyleSheet.create({
  tagContainer: {
    flexDirection: 'row',
    flexWrap: 'wrap',
  },
  tagView: {
    marginBottom: spacing.margin.small,
    marginRight: spacing.margin.xSmall,
  },
  tag: {
    alignSelf: 'baseline',
  },
});

export default SeriesSection;
