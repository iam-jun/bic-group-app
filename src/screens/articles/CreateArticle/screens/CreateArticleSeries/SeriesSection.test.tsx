import React from 'react';

import { act, renderHook, renderWithRedux } from '~/test/testUtils';
import { mockArticle } from '~/test/mock_data/article';
import MockedNavigator from '~/test/MockedNavigator';
import usePostsStore from '~/store/entities/posts';
import { IPost } from '~/interfaces/IPost';
import SeriesSection from './SeriesSection';
import useCreateArticle from '../../hooks/useCreateArticle';

describe('SeriesSection', () => {
  it('should render correctly', () => {
    const article = { ...mockArticle };

    act(() => {
      usePostsStore.getState().actions.addToPosts({ data: article as IPost });
    });
    renderHook(() => useCreateArticle({ articleId: article.id }));

    const wrapper = renderWithRedux(
      <MockedNavigator
        component={() => (
          <SeriesSection
            onPress={() => {
              // do something
            }}
          />
        )}
      />,
    );

    expect(wrapper).toMatchSnapshot();
  });

  it('should disabled if audience is empty', () => {
    const article = { ...mockArticle };
    article.audience.groups = [];

    act(() => {
      usePostsStore.getState().actions.addToPosts({ data: article as IPost });
    });
    renderHook(() => useCreateArticle({ articleId: article.id }));

    const wrapper = renderWithRedux(
      <MockedNavigator
        component={() => (
          <SeriesSection
            onPress={() => {
              // do something
            }}
          />
        )}
      />,
    );

    expect(wrapper).toMatchSnapshot();
  });
});
