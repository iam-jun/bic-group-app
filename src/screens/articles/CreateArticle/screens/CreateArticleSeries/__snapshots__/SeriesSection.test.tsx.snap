// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SeriesSection should disabled if audience is empty 1`] = `
<RNSScreenStack
  style={
    Object {
      "flex": 1,
    }
  }
>
  <RNSScreen
    collapsable={false}
    forwardedRef={[Function]}
    gestureResponseDistance={
      Object {
        "bottom": -1,
        "end": -1,
        "start": -1,
        "top": -1,
      }
    }
    nativeBackButtonDismissalEnabled={false}
    onAppear={[Function]}
    onDisappear={[Function]}
    onDismissed={[Function]}
    onHeaderBackButtonClicked={[Function]}
    onTransitionProgress={[Function]}
    onWillAppear={[Function]}
    onWillDisappear={[Function]}
    replaceAnimation="pop"
    stackPresentation="push"
    style={
      Object {
        "bottom": 0,
        "left": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
      }
    }
    swipeDirection="horizontal"
  >
    <RNSScreenStackHeaderConfig
      backgroundColor="rgb(255, 255, 255)"
      color="rgb(0, 122, 255)"
      hidden={false}
      onAttached={[Function]}
      onDetached={[Function]}
      title="MockedScreen"
      titleColor="rgb(28, 28, 30)"
      topInsetEnabled={true}
      translucent={false}
    />
    <View
      stackPresentation="push"
      style={
        Array [
          Object {
            "flex": 1,
          },
          Object {
            "backgroundColor": "rgb(242, 242, 242)",
          },
          undefined,
        ]
      }
    >
      <View>
        <View
          style={
            Object {
              "backgroundColor": "#FFFFFF",
              "elevation": 2,
              "paddingBottom": 24,
              "paddingHorizontal": 16,
              "paddingTop": 12,
              "shadowColor": "rgba(0, 0, 0, 1)",
              "shadowOffset": Object {
                "height": 2,
                "width": 0,
              },
              "shadowOpacity": 0.12,
              "shadowRadius": 1.5,
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "flexDirection": "row",
              }
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "#181C32",
                    "fontFamily": "BeVietnamPro-SemiBold",
                    "fontSize": 17,
                  },
                  Object {
                    "color": "#181C32",
                  },
                  undefined,
                ]
              }
            >
              Series
            </Text>
            <Text
              style={
                Array [
                  Object {
                    "color": "#181C32",
                    "fontFamily": "BeVietnamPro-SemiBold",
                    "fontSize": 17,
                  },
                  Object {
                    "color": "#BAC0DE",
                  },
                  Object {
                    "marginLeft": 4,
                  },
                ]
              }
            >
              (optional)
            </Text>
          </View>
          <View
            style={
              Object {
                "marginTop": 16,
              }
            }
          >
            <View
              style={
                Object {
                  "flexDirection": "row",
                  "flexWrap": "wrap",
                }
              }
            >
              <View
                style={
                  Object {
                    "marginBottom": 8,
                    "marginRight": 6,
                  }
                }
              >
                <View
                  accessible={true}
                  collapsable={false}
                  focusable={true}
                  onClick={[Function]}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    Object {
                      "alignItems": "center",
                      "alignSelf": "baseline",
                      "backgroundColor": "#F1F2F8",
                      "borderRadius": 100,
                      "flexDirection": "row",
                      "height": 28,
                      "justifyContent": "center",
                      "marginRight": 8,
                      "opacity": 1,
                      "paddingLeft": 4,
                      "paddingRight": 8,
                    }
                  }
                  testID="tag.item"
                >
                  <Text
                    numberOfLines={1}
                    style={
                      Array [
                        Object {
                          "color": "#212745",
                          "fontFamily": "BeVietnamPro-Light",
                          "fontSize": 13,
                        },
                        Object {},
                        Object {
                          "color": "#2D3039",
                          "paddingLeft": 4,
                        },
                      ]
                    }
                    testID="tag.label"
                  >
                    Đây là series có 3 bài articles
                  </Text>
                </View>
              </View>
            </View>
          </View>
          <View
            accessibilityState={
              Object {
                "disabled": false,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "bottom": 0,
                "left": 0,
                "opacity": 1,
                "position": "absolute",
                "right": 0,
                "top": 0,
              }
            }
            testID="button_wrapper"
          >
            <View
              style={
                Object {
                  "position": "absolute",
                  "right": 16,
                  "top": 12,
                }
              }
            >
              <View
                style={
                  Array [
                    Object {
                      "alignItems": "center",
                      "backgroundColor": "#F1F2F8",
                      "borderRadius": 12,
                      "height": 24,
                      "justifyContent": "center",
                      "width": 24,
                    },
                    undefined,
                  ]
                }
              >
                <View
                  accessibilityState={
                    Object {
                      "disabled": true,
                    }
                  }
                  accessible={true}
                  collapsable={false}
                  focusable={false}
                  hitSlop={
                    Object {
                      "bottom": 10,
                      "left": 10,
                      "right": 10,
                      "top": 10,
                    }
                  }
                  onClick={[Function]}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    Object {
                      "opacity": 1,
                    }
                  }
                >
                  <View
                    style={
                      Array [
                        Object {
                          "alignItems": "center",
                          "flexDirection": "row",
                        },
                        undefined,
                        Object {
                          "backgroundColor": undefined,
                        },
                      ]
                    }
                  >
                    <View
                      style={
                        Array [
                          undefined,
                          undefined,
                          undefined,
                        ]
                      }
                    >
                      <View />
                    </View>
                  </View>
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </RNSScreen>
</RNSScreenStack>
`;

exports[`SeriesSection should render correctly 1`] = `
<RNSScreenStack
  style={
    Object {
      "flex": 1,
    }
  }
>
  <RNSScreen
    collapsable={false}
    forwardedRef={[Function]}
    gestureResponseDistance={
      Object {
        "bottom": -1,
        "end": -1,
        "start": -1,
        "top": -1,
      }
    }
    nativeBackButtonDismissalEnabled={false}
    onAppear={[Function]}
    onDisappear={[Function]}
    onDismissed={[Function]}
    onHeaderBackButtonClicked={[Function]}
    onTransitionProgress={[Function]}
    onWillAppear={[Function]}
    onWillDisappear={[Function]}
    replaceAnimation="pop"
    stackPresentation="push"
    style={
      Object {
        "bottom": 0,
        "left": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
      }
    }
    swipeDirection="horizontal"
  >
    <RNSScreenStackHeaderConfig
      backgroundColor="rgb(255, 255, 255)"
      color="rgb(0, 122, 255)"
      hidden={false}
      onAttached={[Function]}
      onDetached={[Function]}
      title="MockedScreen"
      titleColor="rgb(28, 28, 30)"
      topInsetEnabled={true}
      translucent={false}
    />
    <View
      stackPresentation="push"
      style={
        Array [
          Object {
            "flex": 1,
          },
          Object {
            "backgroundColor": "rgb(242, 242, 242)",
          },
          undefined,
        ]
      }
    >
      <View>
        <View
          style={
            Object {
              "backgroundColor": "#FFFFFF",
              "elevation": 2,
              "paddingBottom": 24,
              "paddingHorizontal": 16,
              "paddingTop": 12,
              "shadowColor": "rgba(0, 0, 0, 1)",
              "shadowOffset": Object {
                "height": 2,
                "width": 0,
              },
              "shadowOpacity": 0.12,
              "shadowRadius": 1.5,
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "flexDirection": "row",
              }
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "#181C32",
                    "fontFamily": "BeVietnamPro-SemiBold",
                    "fontSize": 17,
                  },
                  Object {
                    "color": "#181C32",
                  },
                  undefined,
                ]
              }
            >
              Series
            </Text>
            <Text
              style={
                Array [
                  Object {
                    "color": "#181C32",
                    "fontFamily": "BeVietnamPro-SemiBold",
                    "fontSize": 17,
                  },
                  Object {
                    "color": "#BAC0DE",
                  },
                  Object {
                    "marginLeft": 4,
                  },
                ]
              }
            >
              (optional)
            </Text>
          </View>
          <View
            style={
              Object {
                "marginTop": 16,
              }
            }
          >
            <View
              style={
                Object {
                  "flexDirection": "row",
                  "flexWrap": "wrap",
                }
              }
            >
              <View
                style={
                  Object {
                    "marginBottom": 8,
                    "marginRight": 6,
                  }
                }
              >
                <View
                  accessible={true}
                  collapsable={false}
                  focusable={true}
                  onClick={[Function]}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    Object {
                      "alignItems": "center",
                      "alignSelf": "baseline",
                      "backgroundColor": "#F1F2F8",
                      "borderRadius": 100,
                      "flexDirection": "row",
                      "height": 28,
                      "justifyContent": "center",
                      "marginRight": 8,
                      "opacity": 1,
                      "paddingLeft": 4,
                      "paddingRight": 8,
                    }
                  }
                  testID="tag.item"
                >
                  <Text
                    numberOfLines={1}
                    style={
                      Array [
                        Object {
                          "color": "#212745",
                          "fontFamily": "BeVietnamPro-Light",
                          "fontSize": 13,
                        },
                        Object {},
                        Object {
                          "color": "#2D3039",
                          "paddingLeft": 4,
                        },
                      ]
                    }
                    testID="tag.label"
                  >
                    Đây là series có 3 bài articles
                  </Text>
                </View>
              </View>
            </View>
          </View>
          <View
            accessibilityState={
              Object {
                "disabled": false,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "bottom": 0,
                "left": 0,
                "opacity": 1,
                "position": "absolute",
                "right": 0,
                "top": 0,
              }
            }
            testID="button_wrapper"
          >
            <View
              style={
                Object {
                  "position": "absolute",
                  "right": 16,
                  "top": 12,
                }
              }
            >
              <View
                style={
                  Array [
                    Object {
                      "alignItems": "center",
                      "backgroundColor": "#F1F2F8",
                      "borderRadius": 12,
                      "height": 24,
                      "justifyContent": "center",
                      "width": 24,
                    },
                    undefined,
                  ]
                }
              >
                <View
                  accessibilityState={
                    Object {
                      "disabled": true,
                    }
                  }
                  accessible={true}
                  collapsable={false}
                  focusable={false}
                  hitSlop={
                    Object {
                      "bottom": 10,
                      "left": 10,
                      "right": 10,
                      "top": 10,
                    }
                  }
                  onClick={[Function]}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    Object {
                      "opacity": 1,
                    }
                  }
                >
                  <View
                    style={
                      Array [
                        Object {
                          "alignItems": "center",
                          "flexDirection": "row",
                        },
                        undefined,
                        Object {
                          "backgroundColor": undefined,
                        },
                      ]
                    }
                  >
                    <View
                      style={
                        Array [
                          undefined,
                          undefined,
                          undefined,
                        ]
                      }
                    >
                      <View />
                    </View>
                  </View>
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </RNSScreen>
</RNSScreenStack>
`;
